Fixes #769 : Mock files getting created in disk when dry-run is enabled

snopan opened the pull request:
Description
-------------

The PR aims to fix an issue with dry-run option when enabled is still creating the mock files in disk. There's an existing [PR](https://github.com/vektra/mockery/pull/775) to resolve this but I can see it's been inactive since April so I thought to start another PR to resolve the comment. If this isn't okay please let me know, I'm still new to contributing to open source.

- Fixes https://github.com/vektra/mockery/issues/769 (issue)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

Version of Golang used when building/testing:
---------------------------------------------

- [ ] 1.11
- [ ] 1.12
- [ ] 1.13
- [ ] 1.14
- [ ] 1.15
- [ ] 1.16
- [ ] 1.17
- [ ] 1.18
- [ ] 1.19
- [ ] 1.20
- [x] 1.22

How Has This Been Tested?
---------------------------

1. Cleared the mocks by running `task mocks.remove` 
2. Added `dry-run: True` to mockery.yaml
3. Ran `task mocks` which should only log and not create the files

Checklist
-----------

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

snopan commented:
Hey @LandonTClipp, I saw the issue was inactive and I thought to give it a try hope it is okay. Feel free to have a review when you're free, thanks.

LandonTClipp reviewed (approved):
Looks great! Thanks for the fix.

LandonTClipp commented:
Looks like something is wrong with the tests, maybe dry-run is getting enabled in the tests when we didn't expect? I'll let you dig into it.

snopan commented:
Yup so the original unit test had `dryRun` set to true while checking that files are created. Instead I've updated the tests to check `dryRun` as both true and false to confirm that files creation depend on that. I've also removed the `wantError` field from the test case as it's not used, is this common that fields are added but not used? If we need that there then I can add it back.

@LandonTClipp if I can get another review that would be great, thanks!

LandonTClipp reviewed (approved):
Wonderful, thank you!
